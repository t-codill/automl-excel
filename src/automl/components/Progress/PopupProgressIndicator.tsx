import { Dialog, DialogType, ProgressIndicator } from "office-ui-fabric-react";
import * as React from "react";

interface IPopupProgressIndicatorProps {
    title?: string;
    description?: string;
    percentComplete?: number;
    onDismiss?(): void;
}

export class PopupProgressIndicator
    extends React.PureComponent<IPopupProgressIndicatorProps> {
    public render(): React.ReactNode {

        return <Dialog
            hidden={false}
            minWidth={800}
            onDismiss={this.props.onDismiss}
            modalProps={{
                isBlocking: !this.props.onDismiss
            }}
            dialogContentProps={{
                type: DialogType.normal,
                title: this.props.title
            }}
        >
            <ProgressIndicator
                description={this.props.description}
                percentComplete={this.props.percentComplete} />
        </Dialog>;
    }
}
