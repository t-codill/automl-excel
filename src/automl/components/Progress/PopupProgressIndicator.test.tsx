import { shallow } from "enzyme";
import * as React from "react";
import { PopupProgressIndicator } from "./PopupProgressIndicator";

describe("PopupProgressIndicator", () => {
    it("should render", () => {
        const wrapper = shallow(<PopupProgressIndicator />);
        expect(wrapper)
            .toMatchSnapshot();
    });
    it("should render with title", () => {
        const wrapper = shallow(<PopupProgressIndicator title="title1" />);
        expect(wrapper)
            .toMatchSnapshot();
    });
    it("should render with loading title", () => {
        const wrapper = shallow(<PopupProgressIndicator title="loading..." />);
        expect(wrapper)
            .toMatchSnapshot();
    });
    it("should render with loading description", () => {
        const wrapper = shallow(<PopupProgressIndicator description="round and round we go" />);
        expect(wrapper)
            .toMatchSnapshot();
    });
    it("should render with the whole nine yards", () => {
        const wrapper = shallow(<PopupProgressIndicator title="loading" description="round and round we go" />);
        expect(wrapper)
            .toMatchSnapshot();
    });
});
