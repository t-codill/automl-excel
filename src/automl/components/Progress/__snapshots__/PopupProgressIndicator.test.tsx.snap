// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupProgressIndicator should render 1`] = `
<StyledWithResponsiveMode
  dialogContentProps={
    Object {
      "title": undefined,
      "type": 0,
    }
  }
  hidden={false}
  minWidth={800}
  modalProps={
    Object {
      "isBlocking": true,
    }
  }
>
  <StyledProgressIndicatorBase />
</StyledWithResponsiveMode>
`;

exports[`PopupProgressIndicator should render with loading description 1`] = `
<StyledWithResponsiveMode
  dialogContentProps={
    Object {
      "title": undefined,
      "type": 0,
    }
  }
  hidden={false}
  minWidth={800}
  modalProps={
    Object {
      "isBlocking": true,
    }
  }
>
  <StyledProgressIndicatorBase
    description="round and round we go"
  />
</StyledWithResponsiveMode>
`;

exports[`PopupProgressIndicator should render with loading title 1`] = `
<StyledWithResponsiveMode
  dialogContentProps={
    Object {
      "title": "loading...",
      "type": 0,
    }
  }
  hidden={false}
  minWidth={800}
  modalProps={
    Object {
      "isBlocking": true,
    }
  }
>
  <StyledProgressIndicatorBase />
</StyledWithResponsiveMode>
`;

exports[`PopupProgressIndicator should render with the whole nine yards 1`] = `
<StyledWithResponsiveMode
  dialogContentProps={
    Object {
      "title": "loading",
      "type": 0,
    }
  }
  hidden={false}
  minWidth={800}
  modalProps={
    Object {
      "isBlocking": true,
    }
  }
>
  <StyledProgressIndicatorBase
    description="round and round we go"
  />
</StyledWithResponsiveMode>
`;

exports[`PopupProgressIndicator should render with title 1`] = `
<StyledWithResponsiveMode
  dialogContentProps={
    Object {
      "title": "title1",
      "type": 0,
    }
  }
  hidden={false}
  minWidth={800}
  modalProps={
    Object {
      "isBlocking": true,
    }
  }
>
  <StyledProgressIndicatorBase />
</StyledWithResponsiveMode>
`;
